### PR TITLE
Fix demos regression

### PR DESCRIPTION
**Pull request by brendanmoore**
Ref: #619 

@mklopets Quick fixes to get the demos working again...

This:
https://github.com/alex3165/react-mapbox-gl/compare/v4...brendanmoore:feature/react-create-context-typescript-update?expand=1#diff-3003f9bd42e62ece8399351cef1f10c7R130
is interesting, I couldn't actually figure out why the call signature types were incompatible so had to resort to `any` to get it working. I think I am missing something.

### CODE REVIEW

File: example/tsconfig.json [hunk 1]
```diff
@@ -13,13 +14,11 @@
     "noUnusedLocals": true,
     "outDir": "./dist",
     "removeComments": true,
+    "skipLibCheck": true,
     "sourceMap": true,
-    "strictNullChecks": true,
+    "strictNullChecks": false, // Prevents: JSX element type 'ReactElement<any> | null' is not a constructor function for JSX elements. Type 'null' is not assignable to type 'ElementClass'
```
Comment (mklopets) [inline example/tsconfig.json]: would be nice to keep `strictNullChecks` on... I don't really understand why this error is there in the first place tbh

File: example/package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
     "html-webpack-plugin": "2.30.1",
     "raw-loader": "0.5.1",
     "ts-loader": "2.3.7",
-    "typescript": "2.5.3",
+    "typescript": "2.9.2",
```
Comment (mklopets) [inline example/package.json]: It's `^3.0.3` on the actual package (I changed it), please make this match

File: example/src/demos/htmlCluster.tsx [hunk 3]
```diff
@@ -127,7 +127,7 @@ class HtmlCluster extends React.Component<Props, State> {
         onMove={this.onMove}
         style={mapContainerStyle}
       >
-        <Cluster ClusterMarkerFactory={this.clusterMarker}>
+        <Cluster ClusterMarkerFactory={this.clusterMarker as any}>
```
Comment (mklopets) [inline example/src/demos/htmlCluster.tsx]: Try this (`git apply dontcast2any.patch`): https://gist.github.com/mklopets/461f206582af8dca0e96ce7a29c56f59

Comment (mklopets) [reply example/src/demos/htmlCluster.tsx]: still requires a cast but a much more reasonable one than to `any`

File: example/tsconfig.json (hunk 1 above)
Comment (brendanmoore) [reply example/tsconfig.json]: It is because of this: https://github.com/Microsoft/TypeScript/issues/11955

**Comment by brendanmoore** [comment]
Thanks for the `ClusterMarkerFactory as any` pointers... It did occur to me that there is some (for a least) easy to mis-read types around the codebase i.e. `Array<React.Component<MarkerProps>>` and `Array<React.ReactElement<MarkerProps>>` i wonder if it would help to export:

```typescript
export type MarkerElement = React.ReactElement<MarkerProps>;
export type MarkerComponent = React.Component<MarkerProps>;
```
Not sure if this is a good idea?

File: example/tsconfig.json (hunk 1 above)
Comment (mklopets) [reply example/tsconfig.json]: That issue was supposedly fixed though(?)

**Comment by mklopets** [comment]
Will look at the other comment tomorrow - not at a computer right now and don't know the code there by heart

**Comment by brendanmoore** [comment]
@mklopets Looks like an `@types/react` update fixed the `strictNullChecks` issue, _should_ be good to go 😬

**Pull request merged**